src/System.Linq.Expressions/src/System/Dynamic/Utils/TypeExtensions.cs

```diff
@@ -19,47 +19,7 @@ internal static class TypeExtensions
         public static MethodInfo GetAnyStaticMethodValidated(this Type type, string name, Type[] types)
         {
             Debug.Assert(types != null);
-            foreach (MethodInfo method in type.GetMethods(BindingFlags.Public | BindingFlags.NonPublic | BindingFlags.Static | BindingFlags.DeclaredOnly))
-            {
-                if (method.Name == name && method.MatchesArgumentTypes(types))
-                {
-                    return method;
-                }
-            }
-
-            return null;
-        }
-
-        /// <summary>
-        /// Returns true if the method's parameter types are reference assignable from
-        /// the argument types, otherwise false.
-        ///
-        /// An example that can make the method return false is that
-        /// typeof(double).GetMethod("op_Equality", ..., new[] { typeof(double), typeof(int) })
-        /// returns a method with two double parameters, which doesn't match the provided
-        /// argument types.
-        /// </summary>
-        private static bool MatchesArgumentTypes(this MethodInfo mi, Type[] argTypes)
-        {
-            Debug.Assert(mi != null);
-            Debug.Assert(argTypes != null);
-
-            ParameterInfo[] ps = mi.GetParametersCached();
-
-            if (ps.Length != argTypes.Length)
-            {
-                return false;
-            }
-
-            for (int i = 0; i < ps.Length; i++)
-            {
-                if (!TypeUtils.AreReferenceAssignable(ps[i].ParameterType, argTypes[i]))
-                {
-                    return false;
-                }
-            }
-
-            return true;
+            return type.GetMethod(name, BindingFlags.Public | BindingFlags.NonPublic | BindingFlags.Static | BindingFlags.DeclaredOnly, null, types, null);
```

Comment:
Okay.

First, any cases that find issues are going to be pretty obscure. I can though find differences easily enough with:

```C#
public class TestClass
{
        // Not very useful, but serves for demonstration
	public static TestClass operator +(TestClass x, ICloneable y) 
	{
		return new TestClass();
	}
	public static TestClass operator +(TestClass x, IEnumerable y)
	{
		return null;
	}
}

Expression.Add(Expression.Constant(new TestClass()), Expression.Constant("abc"));
```

The behaviour here is:
CoreFX now: Picks one of the methods arbitrarily,
CoreFX after PR: Throws `AmbiguousMatchException`
Desktop: Throws `AmbiguousMatchException`.

Even more noticeable is if we change the above to have:

```C#
public class TestClass
{
	public static TestClass operator +(TestClass x, object y)
	{
		return null;
	}
	public static TestClass operator +(TestClass x, string y)
	{
		return new TestClass();
	}
}
```

Here both desktop and the behaviour after this change is to call the second method. With corefx right now it could pick the first method.

It would seem that this difference between desktop and corefx should be considered a bug in corefx, and this PR to be a fix to it.

One thing that remains is that the desktop version calls `MatchesArgumentTypes` and returns null on failure (it also returns null on either argument being null, which can now happen, whereas in corefx currently that's impossible and asserted as such).

I can't think right now of how any non-null result from this call to `GetMethod` could return something that didn't return true from `MatchesArgumentTypes` so maybe that was just overly-defensive.